pkg/R/utils.R

```diff
@@ -259,3 +259,32 @@ joinTaggedList <- function(tagged_list, cnull) {
   lists <- genCompactLists(tagged_list, cnull)
   mergeCompactLists(lists[[1]], lists[[2]])
 }
+
+# Utility function to reduce a key-value list with predicate
+# Used in *ByKey functions
+# param
+#   item key-val pair
+#   keys/vals env of key/value with hashes
+#   pred predicate function
```

Comment:
I feel like the short name `pred` doesn't convey much about what it's trying to do... what do you think about something like `updateOrCreatePred`?